### PR TITLE
Add frontend configuration for blacklist

### PR DESCRIPTION
**Pull request by Prokyonn**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT

#### What's in this PR?

Adds the frontend configuration for the blacklist in sulu 2.0.

### CODE REVIEW

File: Controller/BlacklistItemController.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function cgetAction(Request $request): Response
             $this->view(
                 new ListRepresentation(
                     $listResponse,
-                    'items',
+                    'blacklist',
```
Comment (alexander-schranz) [inline Controller/BlacklistItemController.php]: bc break should be documented in the UPGRADE.md

File: SuluCommunityBundle.php [hunk 2]
```diff
@@ -21,12 +23,19 @@
...
+
     /**
      * {@inheritdoc}
      */
     public function build(ContainerBuilder $container)
     {
-        parent::build($container);
+        $this->buildPersistence(
+            [
+                BlacklistItem::class => 'sulu.model.blacklist.class',
```
Comment (alexander-schranz) [inline SuluCommunityBundle.php]: Think this should be `blacklist_item.class` or?

and when changing to persistence bundle remove the `sulu_community.blacklisting.item_repository` service definition and use the service create by the Persistence Bundle should be something like `sulu.repository.blacklist_item` or so.

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: SuluCommunityBundle.php (hunk 2 above)
Comment (Prokyonn) [reply SuluCommunityBundle.php]: I guess we can then also remove the `sulu_community.blacklisting.user_repository` service definition right ?

Comment (alexander-schranz) [reply SuluCommunityBundle.php]: Sure but that Entity seems to be missing here or?

File: Controller/BlacklistItemController.php (hunk 1 above)
Comment (alexander-schranz) [reply Controller/BlacklistItemController.php]: and this should be `blacklistItems` or `blacklist_items` not sure what we use here normally

File: Admin/CommunityAdmin.php [hunk 3]
```diff
@@ -24,6 +27,11 @@
  */
 class CommunityAdmin extends Admin
 {
+    const SECURITY_CONTEXT = 'sulu.community.blacklist';
```
Comment (alexander-schranz) [inline Admin/CommunityAdmin.php]: `blacklist_item` to be as specific as possible

File: Admin/CommunityAdmin.php [hunk 4]
```diff
@@ -24,6 +27,11 @@
...
+    const LIST_VIEW = 'sulu_community.blacklist';
+    const ADD_FORM_VIEW = 'sulu_community.blacklist.add_form';
+    const EDIT_FORM_VIEW = 'sulu_community.blacklist.edit_form';
```
Comment (alexander-schranz) [inline Admin/CommunityAdmin.php]: We should rename this constants to `BLACKLIST_ITEM_...` or we should rename the class to `BlacklistItemAdmin`

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: SuluCommunityBundle.php (hunk 2 above)
Comment (Prokyonn) [reply SuluCommunityBundle.php]: yep, I've added it

File: UPGRADE.md [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 ## 2.0.0 (unreleased)
 
+### ListRepresentation relation name changed
+The name of the relation inside of the `_embedded` field has been changed from `items` to `blacklist_items`.
```
Comment (alexander-schranz) [inline UPGRADE.md:7]: add empty line before.

Add also upgrade for repository service ids

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Review by alexander-schranz — APPROVED** [review]

**Comment by Prokyonn** [comment]
@alexander-schranz ping

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@Prokyonn needs rebase

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**